### PR TITLE
Fixes #158 & #113: Implement environment variables files and main refactor

### LINKED ISSUE #113: Refactor main
**Is your feature request related to a problem? Please describe.**
Code should be reflecting proposed modifications [here](https://docs.google.com/spreadsheets/d/1TBFR4gtQHnjkyiCjx9rhQJvwUyDHWBQr2oIYo3T3DCA/edit#gid=1723932745)

**Describe the solution you'd like**
Refactor the use case to do so.

### LINKED ISSUE #158: Implement environment variables process for staging/production
**Is your feature request related to a problem? Please describe.**
Many constants used through the project are dependent of either the staging or production environment, therefore they are environment variables.

**Describe the solution you'd like**
Implement environment variables files to ease the process. The solution will include dotenv. An environment template file is to be created and added to the repo. The file will be passed as a parameter.

### PR DESCRIPTION
**Pull request by maximearmstrong**
**Summary:**

Fixes #158: Implement environment variables process for staging/production
Fixes #113: Refactor main.py

This PR main goal is to implement environment variables for staging and production deployment.
Implementing these demanded many modifications to `main.py`, leaving only few modification to do for #113. Because of this, #113 is also fixed by this PR (removing unused arguments and functions).

Implement

- `.env.template` is added to the repo. Every of the variables in this file must be assigned a value in an `.env.FILE`, that is to be passed as argument to `main.py`.

Changes

- `main.py` was modified so that the imports related to the project are made after loading the environment. A new argument `--path_to_env_var` was added. Default is set to `./.env.staging`, but any `.env.FILE` containing all the environment variables may be passed as argument.
- `constants.py` loads their values from the environment. This avoids changing every file using `constants.py`.
- `create_dataset_entity_for_gtfs_metadata.py` and `request_utils.py` were modified to used the env var from `constants.py`
- `test_request_utils.py` and `test_extract_dataset_infos_from_database.py` were modified to reflect the changes

**Expected behavior:** 

Same as before, but an environment variables file must be present in the repo (`.env.staging`), any file containing the environment variables (`.env.FILE`) must be passed as argument otherwise.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `pytest` to make sure you didn't break anything
- [x] Format the title like "<short description of fix and changes>" (for example - "Check for null value before using field")
- [x] Linked all relevant issues
- [ ] ~Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,223 +1,143 @@
...
         default="GTFS",
         help='Type of the datasets to process. Possible values : "GTFS", "GBFS".',
     )
-    group.add_argument(
-        "--download",
-        action="store_const",
-        const=True,
-        help="Download datasets in memory. Usage : --download (-a | -s SPECIFIC)",
+    parser.add_argument(
+        "--path_to_tmp_data",
```
Comment (annaelleduff) [inline main.py:16]: you can use `-` here as it's easier to type, and args parse will take care of making it an `_` for the attribute name

File: utilities/constants.py [hunk 2]
```diff
@@ -1,43 +1,54 @@
...
+
+# Credentials
+USERNAME = os.environ.get("USERNAME")
+PASSWORD = os.environ.get("PASSWORD")
+
 # Entity codes from the Mobility database
-GTFS_CATALOG_OF_SOURCES_CODE = "Q78"
-GBFS_CATALOG_OF_SOURCES_CODE = "Q86"
-GTFS_SCHEDULE_DATA_FORMAT = "Q29"
+GTFS_CATALOG_OF_SOURCES_CODE = os.environ.get("GTFS_CATALOG_OF_SOURCES_CODE")
```
Comment (annaelleduff) [inline utilities/constants.py:8]: A thought: 
staging env vars and prod env var will be versionned and committed so having them as env var there seems risky. We should look into using secrets on k8s or something like that.

And, I would maybe separate env var constants and string constants as the file is getting pretty big. What do you think?

**Review by annaelleduff — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (maximearmstrong) [reply main.py:16]: Didn't know that, thanks!

File: utilities/constants.py (hunk 2 above)
Comment (maximearmstrong) [reply utilities/constants.py]: I separated `constants.py` in 2, but let's discuss the best option for the credentials before committing the env files.

File: main.py [hunk 3]
```diff
@@ -77,12 +108,38 @@
...
 
     # Get environment variables
-    sparql_url = os.environ.get("SPARQL_URL")
-    api_url = os.environ.get("API_URL")
+    sparql_url = os.environ.get(SPARQL_URL)
+    api_url = os.environ.get(API_URL)
+
+    # Create the different environment variables dictionaries
+    # needed for the use cases
+    create_dataset_env_vars = {
```
Comment (annaelleduff) [inline main.py:120]: Could you explain what this does?  this dictionary doesn't seem to be used.

Comment (maximearmstrong) [reply main.py:120]: Right. I forgot to delete it. It was previously passed to the `create_dataset_entity_for_gtfs_metadata` use case.

**Review by annaelleduff — APPROVED** [review]

**Pull request merged**